pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -168,12 +170,16 @@ class CollectionViewSet(
     @property
     def _deprecation(self):
         """Return deprecated collecion ids."""
-        repo_version = self._repository_version
-        deprecated = AnsibleCollectionDeprecated.objects.filter(
-            repository_version=repo_version
-        ).values_list("collection_id", flat=True)
-        self.deprecated_collections_context = deprecated  # needed by get__serializer_context
-        return deprecated
+        deprecations = Q()
+        for namespace, name in AnsibleCollectionDeprecated.objects.filter(
+            pk__in=self._distro_content
+        ).values_list("namespace", "name"):
+            deprecations |= Q(namespace=namespace, name=name)
+        collection_pks = []
+        if len(deprecations):
+            collection_pks = Collection.objects.filter(deprecations).values_list("pk", flat=True)
```

Comment:
@rochacbruno I queried all deprecations at latest_repo_version.content,
got all namespaces and names and filtered the collections with that